### PR TITLE
Fully Remove Node 10

### PR DESCRIPTION
**Pull request by jason-fox**
**Only after the 1.5.0 release**, Node 10 can be fully removed for the next release. No rush to merge this, just raising it whilst I remember to do it.

### CODE REVIEW

**Comment by fgalan** [comment]
Some comments to take into account before merging (originally posted in https://github.com/telefonicaid/iotagent-json/pull/530):

* Don't remove the v10 unit test, but add this note on them: `# FIXME: node v10 has been deprecated, but we are going to keep its regression for a while`
* File docs/deprecated.md should be updated
* CHANGES_NEXT_RELEASE should include:

```
- Upgrade NodeJS version from 10 to 12 in Dockerfile due to Node 8 End-of-Life
- Set Nodejs 12 as minimum version in packages.json (effectively removing Nodev10 from supported versions)
```

**Comment by fgalan** [comment]
> **Only after the 1.5.0 release**, Node 10 can be fully removed for the next release. No rush to merge this, just raising it whilst I remember to do it.

1.5.0 has been released. This PR could be prepared for merging.

**Comment by jason-fox** [comment]
> Some comments to take into account before merging 

Fixed 14bb84a

File: docs/deprecated.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ The following table provides information about the last sigfox-iotagent version
...
 | --------------------- | ----------------------------------------------- | ----------------------------- |
 | NGSIv1 API            | Not yet defined                                 | Not yet defined               |
-| Support to Node.js v4 | 1.0.0                                           | June 13th, 2018                   |
-| Support to Node.js v6 | 1.0.0                                           | June 13th, 2018                   |
-| Support to Node.js v8 | 1.4.0                                           | April 8th, 2020                     |
-| Support to Node.js v10 | Not defined but it will completed before May 2021 | Not yet defined               |
+| Support to Node.js v4 | 1.0.0                                           | June 13th, 2018               |
+| Support to Node.js v6 | 1.0.0                                           | June 13th, 2018               |
+| Support to Node.js v8 | 1.4.0                                           | April 8th, 2020               |
+| Support to Node.js v10 | 1.5.0                                          | March 3rd, 2021               |
```
Comment (fgalan) [inline docs/deprecated.md:44]: There is also a line above:

```
-   Support to Node.js v10 in sigfox-iotagent 1.5.0. The use of Node.js v12 is highly recommended.
```

which should be changed in a similar way than other IOTAs, i.e.:

```
-   Support to Node.js v10 in sigfox-iotagent 1.5.0 (finally removed in 1.6.0). The use of Node.js v12 is highly recommended.
- ```

Comment (jason-fox) [reply docs/deprecated.md:46]: Fixed b62a4f4

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**